src/SignerAlgorithm.ts

```diff
@@ -1,49 +1,25 @@
-import { Signer, SignerAlgorithm } from './JWT'
-import { EcdsaSignature, fromJose, toJose } from './util'
+import { SignerAlgorithm } from './JWT'
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function instanceOfEcdsaSignature(object: any): object is EcdsaSignature {
-  return typeof object === 'object' && 'r' in object && 's' in object
-}
-
-export function ES256KSignerAlg(recoverable?: boolean): SignerAlgorithm {
-  return async function sign(payload: string, signer: Signer): Promise<string> {
-    const signature: EcdsaSignature | string = await signer(payload)
-    if (instanceOfEcdsaSignature(signature)) {
-      return toJose(signature, recoverable)
-    } else {
-      if (recoverable && typeof fromJose(signature).recoveryParam === 'undefined') {
-        throw new Error(`not_supported: ES256K-R not supported when signer doesn't provide a recovery param`)
-      }
-      return signature
-    }
-  }
-}
+import * as ES256KSignerAlg from './SignerAlg/ES256KSignerAlg'
```

Comment:
These imports can be rephrased as:
```suggestion
import { ES256KSignerAlg } from './SignerAlg/ES256KSignerAlg'
```

and then the code below is simplified as well

Reply:
Understandable. It might have a bit to do with how my mind was working. I grasped things as, and after, I worked through them. I realize it was a huge refactor. It seemed necessary as otherwise the files were getting to get huge combining multiple signers and it would be hard to follow. I told Joel that I was trying to figure things out for my particular project. I need a remote signer over a radio connection. I see mention of constructing a remote signer. I went ahead and added ES256 and walked through the library so I'd have it if I need it when bringing in https://github.com/decentralized-identity/did-jwt-vc later. I'm also trying to build something like this (https://github.com/ceramicnetwork/key-did-provider-ed25519) but for a remote signer. Yeah, I was kind of aggressive with my commit. I think I have a got to get this done mentality. I've been working on a particular (presently unfunded/hobby/or whatever project) for over a year, and get onto myself for not making enough progress to be successful with it. I'll be patient. Maybe in time I'll discover that I did some work that did not need to be done. I realize other people have to deal with what I put out. Maybe I'm at a point where I can talk intelligently about it, if I did not realize before...

Reply:
no need to apologise :)

The bill of changes will be reduced once you remove `package-lock.json` since this project uses `yarn` instead, and further more after removing `ES256-R` and related code.

It might not be worth trying to adapt the blockchain/cosmos bits since AFAIK there is no use of P-256 on the blockchains involved, only secp256k.

Still, there is very much to review, and it will take very long.